apiservice/apiservice.go

```diff
@@ -0,0 +1,268 @@
+package apiservice
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/grpcutils"
+)
+
+func NewFlowAPIService(accessNodeAddressAndPort flow.IdentityList, timeout time.Duration) (*FlowAPIService, error) {
+	accessClients := make([]access.AccessAPIClient, accessNodeAddressAndPort.Count())
+	for i, identity := range accessNodeAddressAndPort {
+		if identity.NetworkPubKey == nil {
+			clientRPCConnection, err := grpc.Dial(
+				identity.Address,
+				grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+				grpc.WithInsecure(), //nolint:staticcheck
+				backend.WithClientUnaryInterceptor(timeout))
+			if err != nil {
+				return nil, err
+			}
+
+			accessClients[i] = access.NewAccessAPIClient(clientRPCConnection)
+		} else {
+			tlsConfig, err := grpcutils.DefaultClientTLSConfig(identity.NetworkPubKey)
+			if err != nil {
+				return nil, err
+			}
+
+			clientRPCConnection, err := grpc.Dial(
+				identity.Address,
+				grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+				grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)),
+				backend.WithClientUnaryInterceptor(timeout))
+			if err != nil {
+				return nil, err
+			}
+
+			accessClients[i] = access.NewAccessAPIClient(clientRPCConnection)
+		}
+	}
+
+	ret := &FlowAPIService{}
+	ret.upstream = accessClients
+	ret.LocalCache = nil
+	ret.roundRobin = 0
+	return ret, nil
+}
+
+type FlowAPIService struct {
+	access.UnimplementedAccessAPIServer
```

Comment:
same comment here about embedding `access.AccessAPIServer` from `LocalCache` instead of reimplementing all of the methods that should be served locally

Reply:
Please review the latest: https://github.com/onflow/flow-go/pull/2370/commits/61fae7fc15bed9aa3969d55ea1af704428ddc9ef
I think it does what you needed already.